force-app/main/default/classes/TestRunProcessor.cls

```diff
@@ -107,7 +95,44 @@ public with sharing class TestRunProcessor implements Schedulable {
                 )
             );
         }
-
+        processPassFailDates(results);
         insert results;
     }
-}
+
+    private void processPassFailDates(List<Test_Run_Method_Result__c> result) {
+        Map<String, List<Test_Run_Method_Result__c>> previousResultsMap = new Map<String, List<Test_Run_Method_Result__c>>();
+        List<Test_Run_Method_Result__c> failedTests = new List<Test_Run_Method_Result__c>();
+        for (Test_Run_Method_Result__c r : result) {
+            if (!r.Method_Pass__c) {
+                failedTests.add(r);
+                previousResultsMap.put(r.Method_Name__c, new List<Test_Run_Method_Result__c>());
+            }
+        }
+        List<Test_Run_Method_Result__c> previousResults = [
+            SELECT Id, Method_Name__c, Method_Pass__c, CreatedDate
+            FROM Test_Run_Method_Result__c
+            WHERE Method_Name__c IN :previousResultsMap.keySet()
+            ORDER BY CreatedDate DESC
+        ];
+
+        for (Test_Run_Method_Result__c f : previousResults) {
+            if (previousResultsMap.containsKey(f.Method_Name__c)) {
+                List<Test_Run_Method_Result__c> tempList = previousResultsMap.get(f.Method_Name__c);
+                tempList.add(f);
+            }
+        }
```

Comment:
unused

Reply:
@veritascs which line? This block  is populating the previous results map

Reply:
Word, not unused! My bad.